src/ui/BibleReferenceSettingTab.ts

```diff
@@ -140,88 +142,67 @@ export class BibleReferenceSettingTab extends PluginSettingTab {
+          )
+          .onChange(async (value) => {
+            this.plugin.settings.calloutDefaultFold = value as CalloutFoldFormat
+            console.debug('Callout Default Folding To: ' + value)
+            await this.plugin.saveSettings()
+            new Notice('Callout Default Folding Updated')
           })
-      )
+      })
   }
 
-  SetUpChapterTagging = (containerEl: HTMLElement): void => {
+  SetUpHeaderFormatOptions = (containerEl: HTMLElement): void => {
     new Setting(containerEl)
-      .setName('Create Chapter Tags')
-      .setDesc('Makes hidden #{book_chapter} tag')
-      .addToggle((toggle) =>
-        toggle
-          .setValue(!!this.plugin.settings?.chapterTagging)
-          .onChange((value) => {
-            this.plugin.settings.chapterTagging = value
-            this.plugin.saveData(this.plugin.settings)
+      .setName('Header Formatting String')
+      .setDesc('Describes how to format the header of the callout.')
+      .addText((text) => 
+        text
+          .setPlaceholder('')
+          .setValue(this.plugin.settings.headFormatString)
+          .onChange(async (value) => {
+            this.plugin.settings.headFormatString = value;
+            console.debug('Header Format String To:' + value)
+            await this.plugin.saveSettings()
           })
       )
   }
 
-  SetUpBookBacklinking = (containerEl: HTMLElement): void => {
+  SetUpFooterFormatOptions = (containerEl: HTMLElement): void => {
     new Setting(containerEl)
-      .setName('Create Book Backlink')
-      .setDesc('Makes [[{book}]] link')
-      .addToggle((toggle) =>
-        toggle
-          .setValue(!!this.plugin.settings?.bookBacklinking)
-          .onChange((value) => {
-            this.plugin.settings.bookBacklinking = value
-            this.plugin.saveData(this.plugin.settings)
+      .setName('Footer Formatting String')
+      .setDesc('Describes how to format the footer of the callout.')
+      .addText((text) => 
+        text
+          .setPlaceholder('')
+          .setValue(this.plugin.settings.tailFormatString)
+          .onChange(async (value) => {
+            this.plugin.settings.tailFormatString = value;
+            console.debug('Footer Format String To:' + value)
+            await this.plugin.saveSettings()
           })
       )
   }
 
-  SetUpChapterBacklinking = (containerEl: HTMLElement): void => {
-    new Setting(containerEl)
-      .setName('Create Chapter Backlink')
-      .setDesc('Makes [[{book_chapter}]] link')
-      .addToggle((toggle) =>
-        toggle
-          .setValue(!!this.plugin.settings?.chapterBacklinking)
-          .onChange((value) => {
-            this.plugin.settings.chapterBacklinking = value
-            this.plugin.saveData(this.plugin.settings)
-          })
-      )
+  SetUpMigrationButton = (containerEl: HTMLElement): void => {
+    let warning = containerEl.createEl('p', { text: 'WARNING: THIS WILL OVERRIDE YOUR CURRENT HEADER AND FOOTER SETTINGS' })
```

Comment:
to clarify, this is a broken change, right?

Reply:
and after migration, both old settings, and new settings ui existed in the setting page.

Is it?

![Uploading image.png…]()

Reply:
I personally feel this is a bit confusing? of this migration warning.

![image](https://github.com/tim-hub/obsidian-bible-reference/assets/2884320/d510344a-b2a7-4e28-bd46-161c67b02c24)

If it is a safe migration, I guess it is OK to just migrate it by default, and fully update the setting UI page, just remove the old ones. 

`> [!Bible] {{verse_reference_link}}` I personally not prefer in this way (template) for header and footer separately


If we want to have a template I would rather have an `Advanced UI` section to have a template for the whole rendering in text area, something like 


```
> [!Bible] {{verse_reference_link}}

{{verses}}

{{chapter}}{{chapterNumber}}{{verseStart}}

```

Reply:
> I personally feel this is a bit confusing? of this migration warning.
> 
> ![image](https://user-images.githubusercontent.com/2884320/261832081-d510344a-b2a7-4e28-bd46-161c67b02c24.png)
> 
> If it is a safe migration, I guess it is OK to just migrate it by default, and fully update the setting UI page, just remove the old ones.
> 
> `> [!Bible] {{verse_reference_link}}` I personally not prefer in this way (template) for header and footer separately
> 
> If we want to have a template I would rather have an `Advanced UI` section to have a template for the whole rendering in text area, something like
> 
> ```
> > [!Bible] {{verse_reference_link}}
> 
> {{verses}}
> 
> {{chapter}}{{chapterNumber}}{{verseStart}}
> ```
Firstly, thanks so much for the contribution. I have some different thoughts about the UI change.


In sum,  what I want to say is.

IMHO, I will suggest.

- Add an `Advance UI` section for template, instead of two for header, footer
- Use Text Area for template for whole bible verses content
- Not make broken changes, mapping the new template to the old structure of settings, (and remove the broken change warning)

Reply:
> Add an Advance UI section for template, instead of two for header, footer

I didn't see a _super_ clean way to do this, due to the way verses are formatted.  Definitely something I can look into though.  Formatting verses is a harder thing, as it isn't as simple as `replaceAll` on a string, however I have a few thoughts about the path forward for this (and admittedly, I eventually wanted to get there anyways).

> Use Text Area for template for whole bible verses content

Missed that - I'll get it when I next go over

> Not make broken changes, mapping the new template to the old structure of settings, (and remove the broken change warning)

The age old problem of migration...
Really, what I want to avoid is having two ways to do any one thing.  In my opinion the template is much clearer about what will go where.  Also, there's no easy way to have both.  Either you're formatting by the settings or you're formatting by the template, which just adds another boolean option to the settings. (or the settings update the template, which is somewhat annoying because it would completely overwrite settings if you changed a value)

I believe the middle ground here would be automatically migrating settings into the new system.  This way, no one should notice a break, but everyone can be on the new system, reducing code complexity.  I wasn't entirely sure how to go about this, but I will attempt some things in my next pass.

We still need a way to manually trigger a migration, was it to not automatically trigger for someone.  The migration button could become a command - I believe that would be more sensible for hiding it from the user unless they want it.  It'll still have a warning, but it'll feel more complete and professional in the context of a entire window prompt I believe.

At the end of all this, I do want to say I believe the average user of this plugin should be able to understand the templating syntax, as it is highly similar to the inbuilt templates plugin (on purpose).  Whether this change is a issue for users should be completely up to how breaking it is, and if it will unexpectedly cause people to lose their formatting.

Reply:
On second thought, with respect to the possible issue of onboarding users into the templating system, a dialogue could be provided that allows you to build your template string.  Basically just a configurator that provides the most common options (such as the ones provided before) and inserts a template string based on those options into your settings.

Reply:
I agree, I dislike the two ways of doing one thing as well, 

But this new UI contained two ways of doing one thing.

![](https://user-images.githubusercontent.com/2884320/261832059-322bc8f1-64f2-40fd-9054-eb1ee3ef44c2.png)


Both the new template and old dropdown can config header and bottom.


This is a big UI change for settings. At least we should not show both template settings, and old header setting in one view  at the same time, it need to be splitted, one need to be hided (through toggle or something) at least, to avoid this UI confusion.